src/modules/ui/configs/handle_routing.js

```diff
@@ -7,29 +7,50 @@ export function changeUrl(reduxStore) {
   // Do not change the URL if we are inside a popState event.
   if (config.insidePopState) return;
 
-  const { api } = reduxStore.getState();
+  const { api, shortcuts } = reduxStore.getState();
   if (!api) return;
 
   const { selectedKind, selectedStory } = api;
   const queryString = qs.stringify({ selectedKind, selectedStory });
 
   if (queryString === '') return;
 
-  const url = `?${queryString}`;
+  const { goFullScreen, showDownPanel, showLeftPanel } = shortcuts;
```

Comment:
You don't need to introduce a layout param which shows some numbers. Simply put readable queryParams in the URL. Like, goFullScreen=1. 

Then it's easy to modify them manually as well.